tests/unit/classifier/svm/LibLinear_unittest.cc

```diff
@@ -16,1310 +15,456 @@
-	train_data(6,0)=3;	train_data(6,1)=1;
-	train_data(7,0)=3;	train_data(7,1)=2;
-	train_data(8,0)=3;	train_data(8,1)=3;
-	train_data(9,0)=4;	train_data(9,1)=2;
-
-	test_data(0,0)=-1;	test_data(1,0)=1;
-	test_data(0,1)=-1;	test_data(1,1)=0;
-	test_data(0,2)=-1;	test_data(1,2)=-1;
-	test_data(0,3)=-2;	test_data(1,3)=0;
-	test_data(0,4)=0;	test_data(1,4)=-1;
-	test_data(0,5)=2;	test_data(1,5)=2;
-	test_data(0,6)=3;	test_data(1,6)=1;
-	test_data(0,7)=3;	test_data(1,7)=2;
-	test_data(0,8)=3;	test_data(1,8)=3;
-	test_data(0,9)=4;	test_data(1,9)=2;
-
-	train_feats = new CDenseFeatures<float64_t>(train_data);
-	test_feats = new CDenseFeatures<float64_t>(test_data);
-
-	SGVector<float64_t> labels(num_samples);
-	for (index_t i = 0; i < num_samples; ++i)
-	{
-		labels[i] = (i < num_samples/2) ? 1.0 : -1.0;
-	}
+		auto ll = new CLibLinear();
+		SG_REF(ll);
 
-	ground_truth = new CBinaryLabels(labels);
+		auto eval = new  CContingencyTableEvaluation();
+		SG_REF(eval);
 
-	SG_REF(ground_truth);
-	SG_REF(train_feats);
-	SG_REF(test_feats);
-}
+		ll->set_bias_enabled(biasEnable);
+		ll->set_features(train_feats);
+		if (C_value)
+				ll->set_C(0.1,0.1); //Only in the case of L2R_L1LOSS_SVC_DUAL
+		ll->set_labels(ground_truth);
+		ll->set_liblinear_solver_type(liblinear_solver_type);
+		ll->train();
 
-//Generate Data for L2 regularized
-void generate_data_l2_simple(CDenseFeatures<float64_t>* &train_feats,
-					 CDenseFeatures<float64_t>* &test_feats,
-					 CBinaryLabels* &ground_truth)
-{
-	index_t num_samples = 10;
-		SGMatrix<float64_t> train_data(2, 10);
-		SGMatrix<float64_t> test_data(2, 10);
-
-		train_data(0,0)=-1;	train_data(1,0)=1;
-		train_data(0,1)=-1;	train_data(1,1)=0;
-		train_data(0,2)=-1;	train_data(1,2)=-1;
-		train_data(0,3)=-2;	train_data(1,3)=0;
-		train_data(0,4)=0;	train_data(1,4)=-1;
-		train_data(0,5)=2;	train_data(1,5)=2;
-		train_data(0,6)=3;	train_data(1,6)=1;
-		train_data(0,7)=3;	train_data(1,7)=2;
-		train_data(0,8)=3;	train_data(1,8)=3;
-		train_data(0,9)=4;	train_data(1,9)=2;
-
-		test_data(0,0)=-1;	test_data(1,0)=1;
-		test_data(0,1)=-1;	test_data(1,1)=0;
-		test_data(0,2)=-1;	test_data(1,2)=-1;
-		test_data(0,3)=-2;	test_data(1,3)=0;
-		test_data(0,4)=0;	test_data(1,4)=-1;
-		test_data(0,5)=2;	test_data(1,5)=2;
-		test_data(0,6)=3;	test_data(1,6)=1;
-		test_data(0,7)=3;	test_data(1,7)=2;
-		test_data(0,8)=3;	test_data(1,8)=3;
-		test_data(0,9)=4;	test_data(1,9)=2;
+		auto pred = ll->apply_binary(test_feats);
+		SG_REF(pred);
 
-		train_feats = new CDenseFeatures<float64_t>(train_data);
-		test_feats = new CDenseFeatures<float64_t>(test_data);
+		auto liblin_accuracy = eval->evaluate(pred, ground_truth);
 
-		SGVector<float64_t> labels(num_samples);
-		for (index_t i = 0; i < num_samples; ++i)
+		if (!biasEnable)
 		{
-			labels[i] = (i < num_samples/2) ? 1.0 : -1.0;
+			for(int i=0;i<t_w.vlen;i++)
+				EXPECT_NEAR(ll->get_w()[i], t_w[i], 1e-5);
 		}
+		else
+		{
+			for(int i=0;i<t_w.vlen;i++)
```

Comment:
`for (auto i : range(t_w.vlen))` ... but this is minor

Reply:
Changed it.